java/client/src/test/java/glide/api/StreamCommandsTest.java

```diff
@@ -0,0 +1,204 @@
+                                        .id("id")
+                                        .makeStream(Boolean.TRUE)
+                                        .trim(new MaxLen(5L, 10L))
+                                        .build(),
+                                new String[] {
+                                    TRIM_MAXLEN_REDIS_API,
+                                    TRIM_EXACT_REDIS_API,
+                                    Long.toString(5L),
+                                    TRIM_LIMIT_REDIS_API,
+                                    Long.toString(10L),
+                                    "id"
+                                }),
+                        Pair.of(
+                                // MAXLEN with non exact match
+                                StreamAddOptions.builder()
+                                        .makeStream(Boolean.FALSE)
+                                        .trim(new MaxLen(false, 2L))
+                                        .build(),
+                                new String[] {
+                                    NO_MAKE_STREAM_REDIS_API,
+                                    TRIM_MAXLEN_REDIS_API,
+                                    TRIM_NOT_EXACT_REDIS_API,
+                                    Long.toString(2L),
+                                    "*"
+                                }),
+                        Pair.of(
+                                // MIN ID with LIMIT
+                                StreamAddOptions.builder()
+                                        .id("id")
+                                        .makeStream(Boolean.TRUE)
+                                        .trim(new MinId("testKey", 10L))
+                                        .build(),
+                                new String[] {
+                                    TRIM_MINID_REDIS_API,
+                                    TRIM_EXACT_REDIS_API,
+                                    Long.toString(5L),
+                                    TRIM_LIMIT_REDIS_API,
+                                    Long.toString(10L),
+                                    "id"
+                                }),
+                        Pair.of(
+                                // MIN ID with non exact match
+                                StreamAddOptions.builder()
+                                        .makeStream(Boolean.FALSE)
+                                        .trim(new MinId(false, "testKey"))
+                                        .build(),
+                                new String[] {
+                                    NO_MAKE_STREAM_REDIS_API,
+                                    TRIM_MINID_REDIS_API,
+                                    TRIM_NOT_EXACT_REDIS_API,
+                                    Long.toString(5L),
+                                    "*"
+                                })));
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
```

Comment:
nit: add third argument to the function - test name - and use it there
```suggestion
    @ParameterizedTest(name = "{0}")
```
(the index is zero based)